datasources/src/commonTest/kotlin/com/mirego/trikot/datasources/BaseHotDataSourceTests.kt

```diff
@@ -158,23 +174,29 @@ class BaseHotDataSourceTests {
                 readPromise
             }
         }
-
-        override fun delete(cacheableId: Any) {
-        }
     }
 
     class CacheDataSource(private val readPromise: Promise<DataSourceTestData>) : BaseHotDataSource<TestDataSourceRequest, DataSourceTestData>() {
         var internalSaveCount = 0
+        var internalDeleteCount = 0
+        var deletedCacheableIds = mutableSetOf<Any>()
 
         override fun internalRead(request: TestDataSourceRequest): Promise<DataSourceTestData> {
-            return readPromise
+            return if (deletedCacheableIds.contains(request.cacheableId)) {
+                Promise.reject(Throwable())
```

Comment:
Maybe reject a Throwable with a clear explanation here? Let's say a custom `RequestCacheDeletedException : Exception("The cache was deleted for this request: $cacheableId")`

Reply:
:this:

Reply:
Done 👍